src/LLMService.py

```diff
@@ -199,308 +85,25 @@ def investigate_issue(self, context: str, issue: Issue) -> tuple[AnalysisRespons
-            HumanMessagePromptTemplate.from_template(self.recommendations_prompt)
-        ])
-
-        try:
-            recommendation_prompt_chain = (
-                {
-                    "cve_error_trace": RunnableLambda(lambda _: issue.trace),
-                    "analysis": RunnableLambda(lambda _: analysis_response.justifications),
-                    "context": RunnableLambda(lambda _: context),
-                }
-                | recommendations_prompt
-            )
-            recommendations_response = robust_structured_output(llm=self.main_llm,
-                                                                schema=RecommendationsResponse,
-                                                                input={},
-                                                                prompt_chain=recommendation_prompt_chain,
-                                                                max_retries=self.max_retry_limit
-                                                                )
-            logger.debug(f"recommendations_response: {recommendations_response=}")
-
-        except Exception as e:
-            logger.error(RED_ERROR_FOR_LLM_REQUEST.format(max_retry_limit=self.max_retry_limit, function_name="_recommand", issue_id=issue.id, error=e))
-            raise e
-
-        return recommendations_response
-
-
-
-    @retry(stop=stop_after_attempt(2),
-           wait=wait_fixed(10),
-           retry=retry_if_exception_type(Exception)
-           )
-    def _evaluate(self, actual_prompt, response, issue_id) -> EvaluationResponse:
-        from langchain_core.prompts import ChatPromptTemplate
-        from langchain_core.runnables import RunnablePassthrough
-
-        prompt = ChatPromptTemplate.from_messages([
-            # Should not use 'system' for deepseek-r1
-            ("user", self.evaluation_prompt)
-        ])
-
-        evaluation_prompt_chain = (
-                {
-                    "actual_prompt": RunnableLambda(lambda _: actual_prompt),
-                    "response": RunnablePassthrough()
-                }
-                | prompt
-        )
-        try:
-            critique_response = robust_structured_output(llm=self.main_llm,
-                                                         schema=EvaluationResponse,
-                                                         input=response,
-                                                         prompt_chain=evaluation_prompt_chain,
-                                                         max_retries=self.max_retry_limit
-                                                        )
-            logger.debug(f"{critique_response=}")
-
-        except Exception as e:
-            logger.error(RED_ERROR_FOR_LLM_REQUEST.format(max_retry_limit=self.max_retry_limit, function_name="_evaluate", issue_id=issue_id, error=e))
-            raise e
-
-        return critique_response
 
     def create_vdb(self, text_data):
```

Comment:
creating vdb is not a responsibility of `llm_service.py` It should be part of the `vector_store_service.py`

Reply:
migrated to vector_store_service.py